packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -0,0 +1,144 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {Result} from '../../api/search/search/result';
+import {isArray, removeDuplicates} from '../../utils/utils';
+import {executeSearch, fetchMoreResults} from '../search/search-actions';
+import {registerFolding} from './folding-actions';
+import {
+  FoldedResult,
+  FoldingFields,
+  getFoldingInitialState,
+} from './folding-state';
+
+export interface ResultWithFolding extends Result {
+  parentResult: ResultWithFolding | null;
+  childResults: ResultWithFolding[];
```

Comment:
When would the `parentResult` be null? And when it is the case, would there still be child results?

Also, are the names of the keys dictated by the api? I think it's strange to have to do e.g.

```
foldedResult.childResults[0].parentResult
```

but maybe that's how the api is designed and we have no choice?

Reply:
that's how it is in the api, we have no say in the type or name of those fields.

`parentResult` is typically null when a result represents the root of a collection.